lib/media/media_source_engine.js

```diff
@@ -215,7 +219,12 @@ shaka.media.MediaSourceEngine = class {
     // Store the object URL for releasing it later.
     this.url_ = shaka.media.MediaSourceEngine.createObjectURL(mediaSource);
 
-    this.video_.src = this.url_;
+    if (this.source_) {
```

Comment:
This should also clear the `src` value on the video element. In case someone plays a src= asset, and then a MediaSource asset, on the same video element.